### PR TITLE
make failed permalink lookups cheaper and silence stage svg 404 error…

### PR DESCRIPTION
**Pull request by grosser**
…s from logs

followup to https://github.com/zendesk/samson/pull/1350

### CODE REVIEW

**Comment by grosser** [comment]
@jonmoter

File: app/models/concerns/permalinkable.rb [hunk 1]
```diff
@@ -9,14 +9,13 @@ module Permalinkable
   end
 
   module ClassMethods
+    # find by permalink or id
+    def find_by_param(param)
+      where("permalink = ? OR id = ?", param, param).first
```
Comment (jonmoter) [inline app/models/concerns/permalinkable.rb]: If you have an `OR` statement like this, does it still take advantage of indices?

**Comment by jonmoter** [comment]
While we're at it, what do you think of adding some caching for the .svg endpoint?  If that endpoint is getting hit over and over again, it seems like a prime candidate for caching.

File: app/models/concerns/permalinkable.rb (hunk 1 above)
Comment (grosser) [inline app/models/concerns/permalinkable.rb]: seems to work fine:

```
=> EXPLAIN for: SELECT `stages`.* FROM `stages` WHERE `stages`.`deleted_at` IS NULL AND (permalink = 'foo' OR id = 'foo') ORDER BY `stages`.`order` ASC
+----+-------------+--------+------+---------------+------+---------+------+------+-----------------------------+
| id | select_type | table  | type | possible_keys | key  | key_len | ref  | rows | Extra                       |
+----+-------------+--------+------+---------------+------+---------+------+------+-----------------------------+
|  1 | SIMPLE      | stages | ALL  | PRIMARY       | NULL | NULL    | NULL | 1948 | Using where; Using filesort |
+----+-------------+--------+------+---------------+------+---------+------+------+-----------------------------+
1 row in set (0.00 sec)

>> Stage.where("permalink = ?", 'foo').explain
=> EXPLAIN for: SELECT `stages`.* FROM `stages` WHERE `stages`.`deleted_at` IS NULL AND (permalink = 'foo') ORDER BY `stages`.`order` ASC
+----+-------------+--------+------+---------------+------+---------+------+------+-----------------------------+
| id | select_type | table  | type | possible_keys | key  | key_len | ref  | rows | Extra                       |
+----+-------------+--------+------+---------------+------+---------+------+------+-----------------------------+
|  1 | SIMPLE      | stages | ALL  | NULL          | NULL | NULL    | NULL | 1948 | Using where; Using filesort |
+----+-------------+--------+------+---------------+------+---------+------+------+-----------------------------+
1 row in set (0.00 sec)
```

Comment (grosser) [inline app/models/concerns/permalinkable.rb]: need to do if/else since non-integers don't work in all databases ...

**Comment by grosser** [comment]
caching is for a different PR ... good to go ?

**Pull request merged**